protocols/protocol.py

```diff
@@ -117,15 +146,122 @@ def extended_validation(self):
                     return False
         return self.validate(self.toJsonDict())
 
-
     @classmethod
-    def validate(cls, jsonDict):
+    def validate(cls, jsonDict, debug=False):
         """
         Validates the specified JSON dictionary to determine if it is an
         instance of this element's schema.
         """
-        return avro.io.validate(cls.schema, jsonDict)
+        if debug:
+            validation_result = ValidationResult()
+            return cls.validate_debug(jsonDict=jsonDict, validation_result=validation_result)
+        return avro.io.validate(expected_schema=cls.schema, datum=jsonDict)
 
+    @classmethod
+    def validate_debug(cls, jsonDict, validation_result, expected_schema=None):
+        """
+        Returns ValidationResult with fields:
+                - result (True or False)
+                - messages (List of message strings ideally to help debug the problem)
+        """
+        INT_MIN_VALUE = -(1 << 31)
+        INT_MAX_VALUE = (1 << 31) - 1
```

Comment:
this is just copy and pasted from `avro.io.validate` so I can only assume they have a slightly different value ?